src/Voron/Data/BTrees/TreeIterator.cs

```diff
@@ -36,19 +42,18 @@ public TreeIterator(Tree tree, LowLevelTransaction tx, bool prefetch)
 
         public int GetCurrentDataSize()
         {
-            if (_disposed)
-                throw new ObjectDisposedException("TreeIterator " + _tree.Name);
+            ThrowIfDisposedOnDebug(this, "TreeIterator " + _tree.Name);
```

Comment:
Do we want to throw only in debug?

Reply:
It wasn't the case in the previous code. @redknightlois?